src/backends/beamcoder.ts

```diff
@@ -242,7 +249,14 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
      * This is the duration of the first video stream in the file expressed in seconds.
      */
     get duration(): number {
-        return this.ptsToTime(this.#demuxer.streams[this.#streamIndex].duration);
+        const stream = this.#demuxer.streams[this.#streamIndex];
+        if (stream.duration !== null) {
+            return this.ptsToTime(stream.duration);
+        }
+        if (stream?.metadata?.DURATION) {
```

Comment:
I don't think it's reliable. it is just a dictionary
https://ffmpeg.org/doxygen/7.0/structAVStream.html#a4e04af7a5a4d8298649850df798dd0bc
Have you tried this?

https://stackoverflow.com/a/27208276
it seems like beamocoder executes similar code under the hood, so, the duration should be available somewhere

Reply:
Note that `stream.duration` is null in this case.

Reply:
Actually, we can call duration on the parent object which seems to return the correct value. I'm going to use that instead.

Reply:
exactly , beamcoder executes code from stack overflow and it expands avformat context